edgelet/iotedge/src/check/additional_info.rs

```diff
@@ -184,3 +195,74 @@ pub(super) fn os_version() -> Result<
         ))
     }
 }
+#[derive(Clone, Debug, serde_derive::Serialize)]
+struct SystemInfo {
+    used_ram: String,
+    total_ram: String,
+    used_swap: String,
+    total_swap: String,
+
+    disks: Vec<DiskInfo>,
+}
+
+impl SystemInfo {
+    fn new() -> Self {
+        let mut system = sysinfo::System::new();
+        system.refresh_all();
+
+        SystemInfo {
+            total_ram: pretty_kbyte(system.get_total_memory()),
+            used_ram: pretty_kbyte(system.get_used_memory()),
+            total_swap: pretty_kbyte(system.get_total_swap()),
+            used_swap: pretty_kbyte(system.get_used_swap()),
+
+            disks: system.get_disks().iter().map(DiskInfo::new).collect(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, serde_derive::Serialize)]
+struct DiskInfo {
+    name: String,
+    percent_free: String,
+    available_space: String,
+    total_space: String,
+    file_system: String,
+    file_type: String,
+}
+
+impl DiskInfo {
+    fn new<T>(disk: &T) -> Self
+    where
+        T: DiskExt,
+    {
+        let available_space = disk.get_available_space();
+        let total_space = disk.get_total_space();
+        #[allow(clippy::cast_precision_loss)]
+        let percent_free = format!(
+            "{:.1}%",
+            available_space as f64 / total_space as f64 * 100.0
+        );
+
+        DiskInfo {
+            name: disk.get_name().to_str().unwrap().to_owned(),
+            percent_free,
+            available_space: Byte::from_bytes(u128::from(available_space))
+                .get_appropriate_unit(true)
+                .format(2),
+            total_space: Byte::from_bytes(u128::from(total_space))
+                .get_appropriate_unit(true)
+                .format(2),
+            file_system: str::from_utf8(disk.get_file_system()).unwrap().to_owned(),
```

Comment:
This will panic if the result of `get_file_system` is not UTF-8.

`file_system: String::from_utf8_lossy(disk.get_file_system()).into_owned(),`